engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -542,21 +545,31 @@ namespace dmGameSystem
                     dmGraphics::GetGraphicsType(attr_model.m_DataType),
                     attr_model.m_Normalize);
 
-                uint8_t* data_write_ptr = scratch_attribute_vertex + custom_vertex_size;
-                memcpy(data_write_ptr, attr_model.m_ValuePtr, attr_model.m_ValueByteSize);
-                custom_vertex_size += attr_model.m_ValueByteSize;
+                uint32_t attribute_index = non_default_attribute.m_NumInfos++;
+                non_default_attribute.m_Infos[attribute_index]                 = attr_model;
+                non_default_attribute.m_Infos[attribute_index].m_ElementCount  = attr_material.m_ElementCount;
+                non_default_attribute.m_VertexStride                          += attr_model.m_ValueByteSize;
```

Comment:
The old setup doesn't work when the model has per-vertex attribute data but is not a "default" attribute, for example in the case where the material has an attribute with a color semantic that differs from the default color semantic attribute type. We need to cover both cases when the model has color data and when it doesn't.